static/app/components/replays/videoReplayer.tsx

```diff
@@ -45,11 +47,12 @@ export class VideoReplayer {
   private _timer = new Timer();
   private _trackList: [ts: number, index: number][];
   private _isPlaying: boolean = false;
+  private _listeners: RemoveListener[] = [];
   /**
    * _videos is a dict that maps attachment index to the video element.
    * Video elements in this dict are preloaded and ready to be played.
    */
-  private _videos: Record<number, HTMLVideoElement>;
+  private _videos: WeakMap<any, HTMLVideoElement>;
```

Comment:
Use a `WeakMap` here just in case, so that the video refs are not retained and leaked.